R/kobo.R

```diff
@@ -63,21 +44,115 @@ Kobo <- R6::R6Class("Kobo",
+    #' @param body R list or json-like string. A data payload to be sent
+    #' to the server.
+    #' @param version character. Indicates on which API version the request
+    #'  should be executed (available: `v1`, `v2`). Defaults to `v2`.
+    post = function(path, body, version = "v2") {
+      if (version == "v2") {
+        if (path != "imports/"){
+          private$session_v2$post(path = paste0("api/v2/", path), body = body)
+        } else {
+          private$session_v2$post(path = path, body = body)
+        }
+      } else if (version == "v1") {
+        if (checkmate::test_null(private$session_v1)) {
+          usethis::ui_stop("Session for API v1 is not initalized.
+          Please re-initalize the Kobo client with the base_url_v1 argument.")
+        }
+        private$session_v1$post(path = paste0("api/v1/", path), body = body)
+      } else {
+        usethis::ui_stop(
+          "Invalid version. Must be either v1 or v2.
+          Come back in a couple of years."
+        )
+      }
+    },
+
     #' @description
     #' Example method to send a GET request to the `assets` endpoint
     #' (due to default to `v2`, no further specification is needed).
     get_assets = function() {
-      self$get("assets/")
+      res_list <- self$get("assets/")
+    },
+
+    #' @description
+    #' High-level POST request to clone an asset. `assets` endpoint
+    #' (due to default to `v2`, no further specification is needed).
+    #' @param clone_from character. UID of the asset to be cloned.
+    #' @param name character. Name of the new asset.
+    #' @param asset_type character. Type of the new asset. Can be
+    #' "block", "question", "survey", "template".
+    clone_asset = function(clone_from, name, asset_type) {
+      body = list(
+        "clone_from" = clone_from,
+        "name" = name,
+        "asset_type" = asset_type)
+      self$post("assets/", body = body)
+    },
+
+    #' @description
+    #' High-level POST request to deploy an asset.
+    #' `assets/{uid}/deployment/` endpoint (due to
+    #' default to `v2`, no further specification is needed).
+    #' @param uid character. UID of the asset to be deployed.
+    deploy_asset = function(uid) {
+      body = list("active" = "true")
+      endpoint = paste0("assets/",uid,"/deployment/")
+      self$post(endpoint, body = body)
+    },
+
+    #' @description
+    #' High-level POST request to create an empty asset. `assets/` endpoint
+    #' (due to default to `v2`, no further specification is needed).
+    #' @param name character. Name of the new asset.
+    #' @param description character. Optional.
+    #' @param sector character. Optional.
+    #' @param country character. Optional.
+    #' @param share_metadata boolean. Optional.
+    #' @param asset_type character. Type of the new asset. Can be
+    #' "block", "question", "survey", "template".
+    create_asset = function(name,description,sector,
+                            country,share_metadata,asset_type) {
```

Comment:
Default values for sector / country, share_metadata could be considered, to make them optional / empty by default or so. Otherwise users have to set them explicitly every time they call the  method.

Reply:
maybe I can comment this method for now?.. for the first merge

Reply:
Yes, I agree! Commenting it out or even exclude it in total from the current PR, and instead write our current findings into a separate issue. Whatever you prefer :)